File: src/calc_23.py
from random import randint
from time import time

count = 0
TOTAL = 100000
MAX_PEOPLE = 23

start = time()

for _ in range(TOTAL):
    days = []
    for _ in range(MAX_PEOPLE):
        birthday = randint(1, 365)

        if birthday in days:
            count += 1
            break

        days.append(birthday)

end = time() - start

print("Probability of %s" % (count / TOTAL))
print("The calculations took %s seconds" % end)

File: src/swift/calc_23.swift
import Foundation

let TOTAL = 100000
let MAX_PEOPLE = 23
var count = 0

let start = Date()

for _ in 0..<TOTAL {
    var days = Set<Int>()

    for _ in 0..<MAX_PEOPLE {
        let birthday = Int.random(in: 0..<365)

        if days.contains(birthday) {
            count += 1
            break
        }

        days.insert(birthday)
    }
}

let end = Date().timeIntervalSince(start)

print("Probability of \(Double(count) / Double(TOTAL))")
print("The calculations took \(end) seconds")
